### ISSUE TITLE
Allow editing the key of a translation unit/source string

### ISSUE THREAD

**Issue by tipa** [WeblateOrg/weblate#10304]
### Describe the problem

I wish to change the key of a source string. This is easily done in my source files ("Find & replace"), but how can I reflect that change in Weblate? I can add and remove source strings in Weblate, but I cannot rename/change the key of a source string

### Describe the solution you'd like

A new menu item "change key":
![image](https://github.com/WeblateOrg/weblate/assets/2863198/ba484643-34dd-4a9c-8d60-34c5703d68de)

or make the "Key" field of the source string editable:
![image](https://github.com/WeblateOrg/weblate/assets/2863198/d2aa720e-516b-4082-9052-f289561869c4)

### Describe alternatives you've considered

I could create a new string in the Weblate UI, but I would then have to update all translations manually and then remove the old string manually.
I could also re-upload all source files after renaming the key, but this would be a lot of work for many languages and it could mess with the History or Labels of the translations

### Additional context

I created my components by manually uploading the translation source files. I am not using GitHub or similar integrations.

**Comment by tipa** [comment]
Any reason why this won't be considered?

**Comment by tipa** [comment]
Happy to give some more context on why I had this situation a few times already when working with translations for my apps:
- Sometimes I find a typo in the source string key that I want to fix. The string value as well as its translations stay the same.
- I name the keys of my source strings like the source string value. E.g. something like `key: "Dialog_Cancel" - value: "Cancel"`. Occasionally I find a better naming for those texts and I might change "Cancel" to "Abort". I would then also change the key `Dialog_Cancel` or `Dialog_Abort`. I would not want to add a new translation unit as "Cancel" and "Abort" is too similar and the translations for the string would likely still be correct (e.g. both "Abort" and "Cancel" translate to "Abbrechen" in German)

**Comment by nijel** [comment]
I understand the motivation, but in most cases this needs a corresponding change in the source code as well. Therefore, this would be an opt-in feature and only few users would actually use that.

**Comment by Ionys320** [comment]
You're right @nijel, but even adding a translation key often requires a change in the source code. I'm currently renaming a lot of keys by removing them and readding them (because the keys of my project is based on the localisation of the translation in the project (page/component/etc)). 

By the way, if the changes are not too complex, I may work on it.

**Comment by nijel** [comment]
Yes, this would be a useful feature for some use-cases, but we're not going to push it on our own.

What I believe would work:

1. Change key for all units in the database, mark them as new (`pending=True, details= {"add_unit": True}` would do that)
2. Remove the old key from all files. This is currently the tricky part, as you would have to modify the files immediately, as there is no way to postpone this right now until https://github.com/WeblateOrg/weblate/issues/10787 is implemented.

**Comment by Ionys320** [comment]
Regarding the second point, isn't this already the case when using the "Delete string" feature?

**Comment by nijel** [comment]
Yes, it does delete strings in the file immediately, what makes it slow.

**Comment by nijel** [comment]
https://github.com/WeblateOrg/weblate/issues/9729 is just about the same issue, just for bilingual files.

**Comment by nijel** [comment]
Just realized there is also https://github.com/WeblateOrg/weblate/issues/2015, which is only about duplicating existing strings but without removing current one.

**Comment by evilaliv3** [comment]
@nijel. by any chance do you have any update on this ticket or on ways to get around this?

at @globaleaks we have same need to:
- revise a source string due to a typo (very few times)
- revise a source string for reasons of accessibility.

for example are changing "home" in "homepage" and in this change we would like to preserve for the moment all the translations that we have for "home" and use them as translations for "homepage".

is there any way to achieve this?

as a comparison transifex has always implemented such a feature.

thank you!
